lib/collection/src/update_handler.rs

```diff
@@ -174,13 +186,21 @@ impl UpdateHandler {
                             if Self::try_recover(segments.clone(), wal.clone()).is_err() {
                                 continue;
                             }
-                            Self::process_optimization(optimizers.clone(), segments.clone());
+                            let mut handles = blocking_handles.lock();
+                            handles.append(&mut Self::process_optimization(
+                                optimizers.clone(),
+                                segments.clone(),
+                            ));
                         }
                         OptimizerSignal::Operation(operation_id) => {
                             if Self::try_recover(segments.clone(), wal.clone()).is_err() {
                                 continue;
                             }
-                            Self::process_optimization(optimizers.clone(), segments.clone());
+                            let mut handles = blocking_handles.lock();
+                            handles.append(&mut Self::process_optimization(
+                                optimizers.clone(),
+                                segments.clone(),
+                            ));
```

Comment:
```suggestion
                            {
                              let mut handles = blocking_handles.lock();
                              handles.append(&mut Self::process_optimization(
                                  optimizers.clone(),
                                  segments.clone(),
                              ));
                            }
```

release lock before continuing with flush